earthfile2llb/interpreter.go

```diff
@@ -1561,10 +1561,25 @@ func parseParans(str string) (string, []string, error) {
 	return parts[0], parts[1:], nil
 }
 
+// isSafeAsyncBuildArgs is used for "BUILD --build-arg key=value +target" style buildargs
 func isSafeAsyncBuildArgs(args []string) bool {
 	for _, arg := range args {
 		_, v, _ := variables.ParseKeyValue(arg)
-		if strings.HasPrefix(v, "$(") {
+		if strings.HasPrefix(v, "$(") || strings.HasPrefix(v, "\"$(") {
+			return false
+		}
+	}
+	return true
+}
+
+// isSafeAsyncBuildArgsNewStyle is used for "BUILD +target --key=value" style buildargs
+func isSafeAsyncBuildArgsNewStyle(args []string) bool {
```

Comment:
Really minor non-blocking comment:

It might be slightly better to rename this to `isSafeAsyncBuildArgs` and rename the other one `isSafeAsyncBuildArgsDeprecated`

Reply:
Good call, code looks funny when there's a `NewStyle` without any `OldStyle`, then v3 introduces `NewerStyle` :laughing: